src/audio/buffer.c

```diff
@@ -218,10 +218,10 @@ void comp_update_buffer_produce(struct comp_buffer __sparse_cache *buffer, uint3
 	/* return if no bytes */
 	if (!bytes) {
 		buf_dbg(buffer, "comp_update_buffer_produce(), no bytes to produce, source->comp.id = %u, source->comp.type = %u, sink->comp.id = %u, sink->comp.type = %u",
-			dev_comp_id(buffer->source),
-			dev_comp_type(buffer->source),
-			dev_comp_id(buffer->sink),
-			dev_comp_type(buffer->sink));
+			(unsigned int)(buffer->source ? dev_comp_id(buffer->source) : UINT32_MAX),
+			(unsigned int)(buffer->source ? dev_comp_type(buffer->source) : UINT32_MAX),
```

Comment:
```suggestion
			buffer->source ? dev_comp_type(buffer->source) :  (uint32_t) UINT32_MAX,
```
Don't give the wrong impression that `dev_comp_id()` is 64 bits.

Reply:
Fixed in update.